src/app/functions/server/googleAnalytics.test.ts

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { it, expect, jest } from "@jest/globals";
+import { CONST_GA4_MEASUREMENT_ID } from "../../../constants";
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+jest.mock("../../../db/tables/google_analytics_clients", () => {
+  return {
+    getClientIdForSubscriber: jest.fn(() =>
+      Promise.resolve({ client_id: "testClientId1", cookie_timestamp: "1234" }),
+    ),
+  };
+});
+
+jest.mock("./logging", () => {
+  class Logging {
+    error(message: string, details: object) {
+      console.error(message, details);
+    }
+  }
+
+  const logger = new Logging();
+  return {
+    logger,
+  };
+});
+
+beforeEach(async () => {
+  global.fetch = jest.fn().mockResolvedValue({ ok: true });
```

Comment:
If you'd like to fix the type errors here:

```suggestion
  global.fetch = jest.fn<typeof global.fetch>().mockResolvedValue({ ok: true } as Response);
```

Reply:
Hm pretty sure I tried that, I ended up switching over to `spy` instead which seems to work fine locally with the linter and the build. I'll leave this comment open until tests are fully passing though, I'm pretty sure the unit test failures are mocking problems and not implementation problems.

Reply:
Ah OK I tried again and had just mocked it incorrectly, fixed in my branch thanks!